component/src/main/java/com/celements/structEditor/DefaultStructuredDataEditorService.java

```diff
@@ -353,6 +359,26 @@ private String getTranslatedValue(XWikiDocument onDoc,
     }
   }
 
+  private Optional<Integer> getNumberForMultilingual(XWikiDocument cellDoc, XWikiDocument onDoc) {
+    return modelAccess.getFieldValue(cellDoc, FIELD_MULTILINGUAL).toJavaUtil()
+        .filter(isMultilingual -> isMultilingual)
+        .map(isMultilingual -> newXObjFetcher(cellDoc, onDoc)
+            .filter(this::isOfRequestOrDefaultLang)
+            .stream())
+        .orElse(Stream.empty())
+        .map(BaseObject::getNumber)
+        .findFirst();
+  }
+
+  private boolean isOfRequestOrDefaultLang(BaseObject xObj) {
+    String xObjLang = asOptional(xObj.getStringValue("lang"))
+        .orElseGet(() -> xObj.getStringValue("language"));
+    return context.getRequestParameter("language").toJavaUtil().map(ImmutableSet::of)
+        .orElseGet(() -> ImmutableSet.of(context.getDefaultLanguage(xObj.getDocumentReference()),
```

Comment:
I think, we should use context language before falling back on document-defaultLanguage.

Reply:
I agree, hence it is actually implemented this way. Notice the `orElseGet`, which will only be performed if there is no language request `language` set.

Reply:
ok, I thought context.getDefaultLanguage(xxx) gets the defaultLanguage of the given DocumentReference. Checking the implementation I understand, that it only checks the configuration of the space and wiki without including the given document. Maybe it should take a space reference instead. Yet, this is not focus of this implementation. :-)